docs/en/observability/apm/configure/outputs/logstash.asciidoc

```diff
@@ -154,11 +171,12 @@ output {
 }
 ----
```

Comment:
Does this look accurate? Specifically:

* Are the placement _and_ syntax of lines 155-157 correct?
* Is line 158 still necessary or would the correct index be assumed based on [`data_stream_auto_routing`](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-data_stream_auto_routing)?

Reply:
> Are the placement and syntax of lines 155-157 correct?

No, the `mutate` block needs to be in a separate `filter` block, not under `output` - see below.

> Is line 158 still necessary or would the correct index be assumed based on [data_stream_auto_routing](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-data_stream_auto_routing)?

It's not necessary when configuring `data_stream => "true"` in the Elasticsearch output. I have not tested this, but I think this is correct:

```
filter {
  if [@metadata][beat] == "apm-server" {
    if [data_stream][type] == "metrics" {
      mutate {
        update => { "[data_stream][dataset]" => "custom" }
      }
    }
  }
}
output {
  elasticsearch {
    data_stream => "true"
    cloud_id => "${CLOUD_ID}"
    cloud_auth => "${CLOUD_AUTH}"
  }
}
```